flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -115,10 +116,21 @@ protected ShardConsumer(KinesisDataFetcher<T> fetcherRef,
 
 		if (lastSequenceNum.equals(SentinelSequenceNumber.SENTINEL_AT_TIMESTAMP_SEQUENCE_NUM.get())) {
 			String timestamp = consumerConfig.getProperty(ConsumerConfigConstants.STREAM_INITIAL_TIMESTAMP);
-			try {
-				this.initTimestamp = KinesisConfigUtil.initTimestampDateFormat.parse(timestamp);
-			} catch (ParseException e) {
-				this.initTimestamp = new Date((long) (Double.parseDouble(timestamp) * 1000));
+
+			if (consumerConfig.containsKey(ConsumerConfigConstants.STREAM_TIMESTAMP_DATE_FORMAT)) {
```

Comment:
I think for this, it would be nice to be able to just say:

```
String dataFormat = consumerConfig.get(
    ConsumerConfigConstants.STREAM_TIMESTAMP_DATE_FORMAT,
    ConsumerConfigConstants.DEFAULT_STREAM_TIMESTAMP_DATE_FORMAT);
```
and then use whatever rendered format to build the data parser. The logic is much easier to understand that way.

`DEFAULT_STREAM_TIMESTAMP_DATE_FORMAT` would basically be the original format in `KinesisConfigUtil.initTimestampDateFormat`.